lib/restrict_admin_by_ip_middleware.rb

```diff
@@ -0,0 +1,32 @@
+class RestrictAdminByIpMiddleware
+  def initialize(app, allowed_ips)
+    @app = app
+    @allowed_ips = allowed_ips.map { |ip| IPAddr.new(ip) }
+  end
+
+  def call(env)
+    status, headers, body = @app.call(env)
```

Comment:
This should only happen if the request isn't restricted, otherwise the action gets on every request, even restricted ones.

Reply:
Any suggestions on what to do about headers?

Reply:
We can set them explicitly to whatever we feel is appropriate. Anything more than `{"Content-Type" => "text/plain"}` necessary?